docs/architecture/testing/integration.md

```diff
@@ -1,4 +1,15 @@
 (integration-testing)=
 ## Integration tests
 
+Integration tests use the built binaries for the CLI and language runtimes. We have some helpers that allow us to create test scenarios and invoke CLI commands.
+
+* [ProgramTest](https://github.com/pulumi/pulumi/blob/master/pkg/testing/integration/program.go#L853): this is typically the easiest way to write an integration test. `ProgramTest` imports a Pulumi program from the `testdata` directory, and runs a series of CLI commands against it.
```

Comment:
There's a shorthand for GH links which you may have seen `gh-file:<repo>#<path>`. So here I think you could write:

```markdown
* [ProgramTest](gh-file:pulumi#pkg/testing/integration/program.go#L853): this is typically the
```

It's not massively shorter but saves a bit of hardcoding and you get a cute GH icon for it if you are so inclined:

![image](https://github.com/user-attachments/assets/ee62550c-c8a6-4661-9dde-9b3d538abcfa)